core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/DsdeHttpDAO.scala

```diff
@@ -43,11 +43,18 @@ trait DsdeHttpDAO extends LazyLogging {
 
   protected def pipeline[A](implicit um: Unmarshaller[ResponseEntity, A]) = executeRequest[A] _
 
-  protected def when500(throwable: Throwable ): Boolean = {
-    throwable match {
-      case t: RawlsExceptionWithErrorReport => t.errorReport.statusCode.exists(_.intValue/100 == 5)
-      case _ => false
-    }
+  protected def when5xx(throwable: Throwable ): Boolean = DsdeHttpDAO.when5xx(throwable)
+
+
+}
+
+object DsdeHttpDAO {
+
+  private def statusCodePredicate(check: StatusCode => Boolean): Throwable => Boolean = {
+    case t: RawlsExceptionWithErrorReport => t.errorReport.statusCode.exists(check)
+    case _ => false
   }
 
+  def when5xx: Throwable => Boolean = statusCodePredicate(_.intValue/100 == 5)
```

Comment:
So the implementation of this is actually the same as "when500", but the name has been changed to be more accurate?

Reply:
Correct. It confused me with its original name